networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/ActiveEth2P2PNetworkTest.java

```diff
@@ -205,6 +210,66 @@ public void onEpoch_shouldUpdateDiscoveryNetworkForkInfo() {
     verifyNoMoreInteractions(discoveryNetwork);
   }
 
+  @Test
+  @SuppressWarnings("unchecked")
+  public void onEpoch_shouldUpdateDiscoveryNetworkForkInfoForSameEpochBpo() {
+    spec =
+        TestSpecFactory.createMinimalFulu(
+            b ->
+                b.altairForkEpoch(altairForkEpoch)
+                    .bellatrixBuilder(bb -> bb.bellatrixForkEpoch(UInt64.valueOf(3)))
+                    .capellaBuilder(cb -> cb.capellaForkEpoch(UInt64.valueOf(4)))
+                    .denebBuilder(db -> db.denebForkEpoch(UInt64.valueOf(5)))
+                    .electraBuilder(eb -> eb.electraForkEpoch(UInt64.valueOf(6)))
+                    .fuluBuilder(
+                        fb ->
+                            fb.fuluForkEpoch(fuluForkEpoch)
+                                .blobSchedule(List.of(new BlobScheduleEntry(fuluForkEpoch, 64)))));
+    storageSystem = InMemoryStorageSystemBuilder.buildDefault(spec);
+    recentChainData = storageSystem.recentChainData();
+    when(discoveryNetwork.start()).thenReturn(SafeFuture.completedFuture(null));
+    genesis = storageSystem.chainUpdater().initializeGenesis();
+    network = createNetwork();
+    setupForkInfo();
+    assertThat(bpoForkDigest).isEqualTo(fuluForkDigest);
+    // Start network
+    verify(discoveryNetwork, never()).setForkInfo(any(), any(), any(), any(), any());
+    assertThat(network.start()).isCompleted();
+
+    // Verify updates at startup
+    verify(discoveryNetwork).start();
+    ForkInfo expectedFork = new ForkInfo(phase0Fork, genesisValidatorsRoot);
+    final BlobParameters fuluBpo1BlobParameters = new BlobParameters(fuluForkEpoch, 64);
+    verify(discoveryNetwork)
+        .setForkInfo(
+            expectedFork,
+            phase0ForkDigest,
+            Optional.of(altairFork),
+            Optional.of(fuluBpo1BlobParameters),
+            Optional.of(altairForkDigest));
+
+    // Process epoch 1 - we shouldn't update fork info here
+    network.onEpoch(UInt64.ONE);
+    asyncRunner.executeDueActions();
+    // At the altair upgrade epoch, we should update fork info
+    network.onEpoch(altairForkEpoch);
+    // Processing altair again shouldn't cause any updates
+    network.onEpoch(altairForkEpoch);
+    // Reprocessing prior epoch should not update fork info
+    network.onEpoch(UInt64.ONE);
+    reset(discoveryNetwork);
+    // At the fulu upgrade epoch, we should update fork info
+    network.onEpoch(fuluForkEpoch);
+    expectedFork = new ForkInfo(fuluFork, genesisValidatorsRoot);
+    verify(discoveryNetwork)
+        .setForkInfo(
+            expectedFork, fuluForkDigest, Optional.empty(), Optional.empty(), Optional.empty());
+
+    // At the BPO fork upgrade epoch, we should update fork info
```

Comment:
this comment should be `// Processing BPO fork shouldn't cause any updates since it's the same epoch as fulu fork`